src/browser-lib/single-spa-angular.ts

```diff
@@ -39,111 +41,137 @@ export default function singleSpaAngular(userOpts: SingleSpaAngularOpts): LifeCy
-      if (!domElementGetter) {
-        throw Error(`cannot mount angular application '${props.name || props.appName}' without a domElementGetter provided either as an opt or a prop`);
-      }
-
-      const containerEl = getContainerEl(domElementGetter);
-      containerEl.innerHTML = opts.template;
-    })
-    .then(() => {
-      const bootstrapPromise = opts.bootstrapFunction(props)
-      if (!(bootstrapPromise instanceof Promise)) {
-        throw Error(`single-spa-angular: the opts.bootstrapFunction must return a promise, but instead returned a '${typeof bootstrapPromise}' that is not a Promise`);
-      }
-
-      return bootstrapPromise.then(module => {
-        if (!module || typeof module.destroy !== 'function') {
-          throw Error(`single-spa-angular: the opts.bootstrapFunction returned a promise that did not resolve with a valid Angular module. Did you call platformBrowser().bootstrapModuleFactory() correctly?`)
-        }
-        opts.bootstrappedNgZone = module.injector.get(opts.NgZone)
-        opts.bootstrappedNgZone._inner._properties[opts.zoneIdentifier] = true;
-        window.addEventListener('single-spa:routing-event', opts.routingEventListener)
-
-        opts.bootstrappedModule = module;
-        return module;
-      });
+  opts.routingEventListener = () => {
+    opts.bootstrappedNgZone.run(() => {
+      // See https://github.com/single-spa/single-spa-angular/issues/86
+      // Zone is unaware of the single-spa navigation change and so Angular change detection doesn't work
+      // unless we tell Zone that something happened
     });
+  };
+}
+
+async function mount(opts: SingleSpaAngularOpts, props: any): Promise<NgModuleRef<any>> {
+  await Promise.resolve();
+
+  const domElementGetter = chooseDomElementGetter(opts, props);
+  if (!domElementGetter) {
+    throw Error(
+      `cannot mount angular application '${
+        props.name || props.appName
+      }' without a domElementGetter provided either as an opt or a prop`
+    );
+  }
+
+  const containerEl = getContainerEl(domElementGetter);
+  containerEl.innerHTML = opts.template;
+  const bootstrapPromise = opts.bootstrapFunction(props);
+
+  if (!(bootstrapPromise instanceof Promise)) {
+    throw Error(
+      `single-spa-angular: the opts.bootstrapFunction must return a promise, but instead returned a '${typeof bootstrapPromise}' that is not a Promise`
+    );
+  }
+
+  const module: NgModuleRef<any> = await bootstrapPromise;
+
+  if (!module || typeof module.destroy !== 'function') {
+    throw Error(
+      `single-spa-angular: the opts.bootstrapFunction returned a promise that did not resolve with a valid Angular module. Did you call platformBrowser().bootstrapModuleFactory() correctly?`
+    );
+  }
+
+  const singleSpaPlatformLocation: SingleSpaPlatformLocation | null = module.injector.get(
+    SingleSpaPlatformLocation,
+    null
+  );
+
+  if (singleSpaPlatformLocation === null) {
```

Comment:
Hmm do you think it would be best to require these providers to be used by throwing an error if they aren't? I guess there is no situation where someone shouldn't be using the providers?

Reply:
Good point. Since this provider is needed for routing I will add a check `&& opts.Router` (thus the library consumer also has routing in his app)

Reply:
👍